SignalServiceKit/src/Messages/Interactions/OWSContact.m

```diff
@@ -398,6 +407,55 @@ - (NSString *)debugDescription
     return result;
 }
 
+- (OWSContact *)newContactWithNamePrefix:(nullable NSString *)namePrefix
```

Comment:
`OWSContact` is intended to be immutable once we've finished building it.  These methods can be used to "create" and "derive" new contacts.

Reply:
I like that you're working towards making the model less mutable 👍  

But for now we still mutate fields in the `valueField.applyToContact(contact: contact)` methods right?

Reply:
Yeah, it's a grey area.  I consider that "building" the derived contact share, so I allowed it.  We could have been more strict with making these models immutable, but it would have involved significant complexity with IMO little real benefit.